src/test/regress/input/tablespace.source

```diff
@@ -146,3 +146,47 @@ DROP SCHEMA testschema CASCADE;
 
 DROP ROLE tablespace_testuser1;
 DROP ROLE tablespace_testuser2;
+
+-- Create a new tablespace and check that within this tablespace,
+-- after dropping the table, the size of all deleted (but still opened) files is 0.
+CREATE TABLESPACE testspace LOCATION '@testtablespace@';
+DROP TABLE IF EXISTS t;
+
+-- Checking that the AO row-oriented table doesn't leave non-zero files.
+CREATE TABLE t(a int, b int) WITH (appendoptimized = true, orientation=row) TABLESPACE testspace;
+INSERT INTO t SELECT i, i FROM generate_series(1,1000) i;
+DROP TABLE t;
+\! find /proc/[0-9]*/fd -lname '@testtablespace@*(deleted)' -exec stat -Lc '%s' {} \; 2>/dev/null | awk 'BEGIN {sum=0} {sum += $1} END {print "size:",  sum}';
+
+-- Checking that the AO column-oriented table doesn't leave non-zero files.
+CREATE TABLE t(a int, b int) WITH (appendoptimized = true, orientation=column) TABLESPACE testspace;
+INSERT INTO t SELECT i, i FROM generate_series(1,1000) i;
+DROP TABLE t;
+\! find /proc/[0-9]*/fd -lname '@testtablespace@*(deleted)' -exec stat -Lc '%s' {} \; 2>/dev/null | awk 'BEGIN {sum=0} {sum += $1} END {print "size:",  sum}';
+
+-- start_ignore
+-- Enable gp_inject_fault extension.
+-- This extension will be used in test below to force flushing of all dirty
+-- pages by the bg writer process.
+CREATE EXTENSION IF NOT EXISTS gp_inject_fault;
+-- Reset all faults.
+SELECT gp_inject_fault_infinite('all', 'reset', dbid) FROM gp_segment_configuration;
+-- Start with a clean state (no dirty buffers) by performing a checkpoint.
+-- It will also protect from a random timed checkpoint during the test, that could lead to not stable test result.
+CHECKPOINT;
+-- After a checkpoint is requested, need to wait for some time to ensure that the checkpointer process 
+-- completes its work before the actual test starts.
+SELECT pg_sleep(1);
+-- Force flushing all dirty pages by the bg writer process.
+SELECT gp_inject_fault_infinite('bg_buffer_sync_default_logic', 'skip', dbid) FROM gp_segment_configuration WHERE role = 'p' AND content > -1;
```

Comment:
I think we shouldn't make the reset at the begining. Instead, it would be better to reset gp_inject_fault_infinite('bg_buffer_sync_default_logic', 'reset', dbid)  we activated after the test.